src/mountscript.sh

```diff
@@ -602,6 +652,329 @@ ensure_aznfswatchdog()
+        eecho "Failed to add log file path to $stunnel_conf_file!"
+        return 1
+    fi
+
+    stunnel_pid_file="$STUNNELDIR/logs/stunnel_$storageaccount.pid"
+    echo "pid = $stunnel_pid_file" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add pid file path to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo >> $stunnel_conf_file
+
+    echo "[$nfs_host]" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add $nfs_host service/entry name to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "client = yes" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to 'client = yes' to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "accept = $LOCALHOST:$available_port" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add 'accept' info to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "connect = $nfs_host:$CONNECT_PORT" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add 'connect' info to $stunnel_conf_file!"
+        return 1
+    fi
+
+    chattr -f +i $stunnel_conf_file
+}
+
+#
+# Mount nfsv4 files share with TLS encryption.
+#
+tls_nfsv4_files_share_mount()
+{
+    vecho "nfs_dir=[$nfs_dir], mount_point=[$mount_point], options=[$OPTIONS], mount_options=[$MOUNT_OPTIONS]."
+
+    # Note the available port
+    available_port=$(get_next_available_port)
+    if [ $? -ne 0 ]; then
+        eecho "Failed to get the available port for nfsv4 mount."
+        exit 1
+    fi
+
+    vecho "Available Port: $available_port"
+
+    if [ -z "$available_port" ]; then
+        eecho "Running out of ports. Nfsv4 has port range $NFSV4_PORT_RANGE_START to $NFSV4_PORT_RANGE_END. All ports from this range are used by other processes."
+        exit 1
+    fi
+
+    EntryExistinMountMap="true"
+
+    stunnel_conf_file="$STUNNELDIR/stunnel_$storageaccount.conf"
+
+    if [ ! -f $stunnel_conf_file ]; then
+        EntryExistinMountMap="false"
+    fi
+
+    if [ "$EntryExistinMountMap" == "false" ]; then
+        touch $stunnel_conf_file
+        if [ $? -ne 0 ]; then
+            eecho "[FATAL] Not able to create '${stunnel_conf_file}'!"
+            exit 1
+        fi
+
+        chattr -f +i $stunnel_conf_file
+
+        stunnel_log_file=
+        stunnel_pid_file=
+
+        add_stunnel_configuration
+        add_stunnel_configuration_status=$?
+
+        if [ $add_stunnel_configuration_status -ne 0 ]; then
+            eecho "Failed to add stunnel configuration to $stunnel_conf_file!"
+            chattr -i -f $stunnel_conf_file
+            rm $stunnel_conf_file
+            exit 1
+        fi
+
+        # start the stunnel process
+        stunnel_status=$(stunnel $stunnel_conf_file 2>&1)
```

Comment:
name of the process can be a constant?

Reply:
I looked into the code. We are handling the errors slightly differently at all the places. So even if I write a wrapper function to start the stunnel process and handle the errors, the callers will still have to parse the status code and figure out why stunnel failed to start and do the handling. So in my opinion, we won't save much lines by having a common function to start stunnel process.